cmd/execution/main.go

```diff
@@ -564,6 +605,55 @@ func main() {
 		}).Run()
 }
 
+// getContractEpochCounter Gets the epoch counters from the FlowEpoch smart contract from the latest executed block.
+func getContractEpochCounter(vm *fvm.VirtualMachine, vmCtx fvm.Context, executionState state.ExecutionState) (uint64, error) {
+	// Get the address of the FlowEpoch smart contract
+	sc, err := systemcontracts.SystemContractsForChain(vmCtx.Chain.ChainID())
+	if err != nil {
+		return 0, err
+	}
+	address := sc.Epoch.Address
+
+	// Generate the script to get the epoch counter from the FlowEpoch smart contract
+	scriptCode := templates.GenerateGetCurrentEpochCounterScript(templates.Environment{
+		EpochAddress: address.Hex(),
+	})
+	script := fvm.Script(scriptCode)
+
+	// Get latest executed block and a view at that block
+	ctx := context.Background()
+	_, blockID, err := executionState.GetHighestExecutedBlockID(ctx)
+	if err != nil {
+		return 0, err
+	}
+	stateCommit, err := executionState.StateCommitmentByBlockID(ctx, blockID)
+	if err != nil {
+		return 0, err
+	}
+	blockView := executionState.NewView(stateCommit)
+
+	// Create empty programs cache
+	prog := programs.NewEmptyPrograms()
+
+	// execute the script
+	err = vm.Run(vmCtx, script, blockView, prog)
+	if err != nil {
+		return 0, err
+	}
+
+	// Return script results
+	if script.Err == nil && script.Value != nil {
+		epochCounter := script.Value.ToGoValue().(uint64)
+		return epochCounter, nil
+	}
+
+	if script.Err != nil {
+		return 0, script.Err
+	}
+
+	return 0, fmt.Errorf("script returned no value")
```

Comment:
```suggestion
	if script.Err != nil {
		return 0, fmt.Errorf("could not read epoch counter, script error: %w", script.Err)
	}
	
	if script.Value == nil {
		return 0, fmt.Errorf("script to read epoch counter returned no value at block view %v", blockView)
	}

	epochCounter := script.Value.ToGoValue().(uint64)
	return epochCounter, nil
	
```

Reply:
I added the blockId to the errors outside of this function, and refactored a bit.